src/redux/slices/onboarding.ts

```diff
@@ -0,0 +1,84 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from '@/redux/RootState';
+import SliceName from '@/redux/types/SliceName';
+import OnboardingGroup from '@/types/OnboardingGroup';
+
+type ActiveStepPayload = { group: OnboardingGroup; index: number };
+
+export type OnboardingState = {
+  isChecklistVisible: boolean;
+  checklistDismissals: number;
+  lastChecklistDismissal: Date | null;
+  activeStep: ActiveStepPayload;
+  groups: Partial<
+    Record<
+      OnboardingGroup,
+      {
+        completedSteps: number[];
+      }
+    >
+  >;
+};
+
+const initialState: OnboardingState = {
+  isChecklistVisible: true,
+  checklistDismissals: 0,
+  lastChecklistDismissal: null,
+  activeStep: {
+    group: OnboardingGroup.HOMEPAGE,
+    index: 0,
+  },
+  groups: {},
+};
+
+export const onboardingSlice = createSlice({
+  name: SliceName.ONBOARDING,
+  initialState,
+  reducers: {
+    // checklist state
+    dismissChecklist: (
+      state: OnboardingState,
+      action: PayloadAction<{ countDismiss: boolean }>,
+    ) => ({
+      ...state,
+      isChecklistVisible: false,
+      ...(action.payload.countDismiss && {
+        checklistDismissals: state.checklistDismissals + 1,
+        lastChecklistDismissal: new Date(),
```

Comment:
Clicking on next/prev buttons show this error:

<img width="754" alt="Screenshot 2024-01-26 at 1 03 27 am" src="https://github.com/quran/quran.com-frontend-next/assets/15169499/4205f613-b426-4f40-94c7-899e03485fae">